### PR TITLE
Add SCA module to agent simulator

### PR DESCRIPTION
**Pull request by pacoqack**
|Related issue|
|---|
|Closes https://github.com/wazuh/wazuh-qa/issues/1162|

This is a development pull request for internal review. This code will be merged in master as part of this issue: https://github.com/wazuh/wazuh-qa/issues/1133

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py [hunk 1]
```diff
@@ -101,12 +102,13 @@ class Agent:
         syscollector_frequency (int): frequency to run syscollector scans. 0 to continuously send syscollector events.
         keepalive_frequency (int): frequency to send keepalive messages. 0 to continuously send keepalive messages.
         syscollector_batch_size (int): Size of the syscollector type batch events.
+	    sca_frequency (int): frequency to run SCA scans. 0 to continuously send SCA events.
```
Comment (Rebits) [inline deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py:105]:
```suggestion
        syscollector_batch_size (int): Size of the syscollector type batch events.
	  sca_frequency (int): frequency to run SCA scans. 0 to continuously send SCA events.
```

File: deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py [hunk 2]
```diff
@@ -1115,96 +1223,56 @@ def keep_alive(self):
...
-            self.totalMessages += 1
-            if self.totalMessages % self.agent.modules["fim"]["eps"] == 0:
-                sleep(1.0 - ((time() - start_time) % 1.0))
-
-    def fim_integrity(self):
-        """Send an integrity FIM message from the agent to the manager"""
+
+    def run_module(self, module):
+        """Send a module message from the agent to the manager."""
+        module_info = self.agent.modules[module]
```
Comment (Rebits) [inline deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py:1229]:
```suggestion
        """Send a module message from the agent to the manager.
         Args:
                module (str): Module name 
        """
        module_info = self.agent.modules[module]
```

File: deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py [hunk 3]
```diff
@@ -692,12 +703,100 @@ def generate_event(self):
...
+            invalid_checks = total_checks - failed_checks - passed_checks
+            event_data['passed'] = passed_checks
+            event_data['failed'] = failed_checks
+            event_data['invalid'] = invalid_checks
+            event_data['total_checks'] = total_checks
+            event_data['score'] = 20
+            event_data['start_time'] = self.started_time
+            sleep(1)
+            self.started_time = int(time())
+            event_data['end_time'] = self.started_time
```
Comment (Rebits) [inline deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py:751]: Is necessary this sleep?

```suggestion
            event_data['start_time'] = self.started_time
            self.started_time = int(time() + 1)
            event_data['end_time'] = self.started_time
```

**Review by Rebits — COMMENTED** [review]
GJ, but some changes are required.

File: deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py (hunk 3 above)
Comment (pacoqack) [reply deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py:751]: No, it isn't, thank you!

**Comment by pacoqack** [comment]
The suggested changes were applied!

**Review by Rebits — APPROVED** [review]

**Pull request merged**